fieldservice/models/fsm_order.py

```diff
@@ -224,6 +224,16 @@ def write(self, vals):
                 order.customer_id = order.location_id.customer_id.id
         return res
 
+    @api.multi
+    def unlink(self):
+        for order in self:
+            if order.stage_id == order._default_stage_id():
+                res = super(FSMOrder, order).unlink()
```

Comment:
May I suggest to put the test in a seperate function instead of directly in unlink ?
```py
if order.can_unlink():
  res = super().unlink()
else:
  raise ...
```
Otherwise it's hard to change the conditions in inherited classes.